dotnet/src/SemanticKernel/Text/TextChunker.cs

```diff
@@ -129,14 +143,14 @@ private static List<string> InternalSplitTextParagraphs(List<string> lines, int
         return paragraphs;
     }
 
-    private static List<string> BuildParagraph(IEnumerable<string> truncatedLines, int maxTokensPerParagraph, Func<string, int, List<string>> longLinesSplitter)
+    private static List<string> BuildParagraph(IEnumerable<string> truncatedLines, int maxTokensPerParagraph, Func<string, int, List<string>> longLinesSplitter, TokenCounter tokenCounter)
     {
         StringBuilder paragraphBuilder = new();
         List<string> paragraphs = new();
 
         foreach (string line in truncatedLines)
         {
-            if (paragraphBuilder.Length > 0 && TokenCount(paragraphBuilder.Length) + TokenCount(line.Length) + 1 >= maxTokensPerParagraph)
+            if (paragraphBuilder.Length > 0 && tokenCounter(paragraphBuilder.ToString()) + tokenCounter(line) + 1 >= maxTokensPerParagraph)
```

Comment:
nit: You could wrap the delegate with a couple internal methods (see the Encode method support in GPT3Tokenizer).

Reply:
I think the GPT3Tokenizer Encode methods differ slightly in that they're publicly exposed and all call the same base method (not utilized again internally). These internal methods would need to be called quite a bit which would make even basic calls decently verbose?

Something like this...
```
private static int CountTokens(string input, TokenCounter tokenCounter) => tokenCounter(input);
private static int CountTokens(StringBuilder stringBuilder, TokenCounter tokenCounter) => tokenCounter(stringBuilder.ToString());
```

Seems overly complex for the gain.